packages/devtools_shared/lib/src/extensions/extension_model.dart

```diff
@@ -36,6 +36,18 @@ class DevToolsExtensionConfig implements Comparable {
           versionKey: final String version,
           isPubliclyHostedKey: final String isPubliclyHosted,
         }) {
+      final underscoresAndLetters = RegExp(r'[a-z0-9_]*');
```

Comment:
Consider using this expression instead so you don't have to deal with matches on substrings of `name`: `RegExp(r'^[a-z0-9]*$')`

Then you can just do this:

```dart
if (!underscoresAndLetters.hasMatch(name)) {
  throw StateError(
    // ...
  );
}
```